### PR TITLE
ci(jenkins): enable pipeline in apm-ci.elastic.co

### PR DESCRIPTION
**Pull request by v1v**
### What does this PR do?

Enable the CI pipeline in apm-ci.elastic.co. For such it's required the below tasks:
- Configure the `JJBB` to create the UI jobs in the Jenkins CI.
- Configure the CI pipeline to run the required tests for the list of required Node.js versions. Similar to the one already in place for the apm-agent-nodejs repo. See https://github.com/elastic/apm-agent-nodejs/blob/master/.ci/.jenkins_nodejs.yml
- Notify the build status in the README.md.
- Delegate the configuration of the test environments with docker. Therefore, easy to reproduce locally.
- Delete the travis-ci configuration.

### Why is it important?

Align the same CI pipelines for all the projects. Then we can move forward to automate the release process and run downstream projects which are dependent on this one if required.

### Questions
- Do we need to support other platforms (Windows, MacOS)?
- Is the codecov required for this project?
- Is the release process similar to the Node.js agent one?

### Tests

- UI view for the whole pipeline in my local Jenkins instance.
image

- UI view for the test reports in my local Jenkins instance.

image

### CODE REVIEW

File: .ci/scripts/convert_tap_to_junit.sh [hunk 1]
```diff
@@ -0,0 +1,24 @@
+#!/usr/bin/env bash
```
Comment (mdelapenya) [inline .ci/scripts/convert_tap_to_junit.sh]: Is this file a candidate now to be part of the shared library?

Comment (v1v) [reply .ci/scripts/convert_tap_to_junit.sh]: That's a good point, I'll raise an issue to discuss it then

**Comment by v1v** [comment]
@watson , thanks for your comment. I just added the support for the Node.js 6 version.
Regarding keeping both CI tools, I'm afraid we cannot support both tools since Jenkins is the standard one for all the APM/Observability repositories, besides it does allow us to apply an automated release process and Continous Delivery/Deployment with the rest of projects. So in order to keep the same aligned tools strategy, we won't keep Travis in this case.

Regarding how to help the contributors to understand how to navigate it to figure out why their PRs are failing CI, in fact, this is a good point and really valuable, thanks for highlighting it. We do also understand this is something with room to be improved and we have started to work on this. In the following weeks, we will gather with some other team members to understand their use cases and propose some improvements, so I'll keep you posted as your feedback is really valuable.

**Comment by watson** [comment]
I understand. You could consider keeping a single quick "smoke-test" build on Travis, e.g. just the latest LTS version of Node.js and not making it required for merging. Anyway, this is just my 5-cents :)

**Comment by v1v** [comment]
@watson , https://github.com/elastic/apm-agent-nodejs-opentracing/pull/12/commits/60da6aed73a4838b62028adf74a320480145adaa is the one with the smoke test in travis. I just configured the master branch without the requirement of validating the travis build.

File: .ci/.jenkins_nodejs.yml [hunk 2]
```diff
@@ -0,0 +1,13 @@
+NODEJS_VERSION:
+  - "13"
+  - "13.0"
+  - "12"
+  - "12.0"
+  - "11"
+  - "11.0"
+  - "10"
+  - "10.0"
+  - "8"
```
Comment (kuisathaverat) [inline .ci/.jenkins_nodejs.yml]: Node.js 8? on 2020 is on EOL https://nodejs.org/en/about/releases/

File: .ci/.jenkins_nodejs.yml [hunk 3]
```diff
@@ -0,0 +1,13 @@
...
+  - "8.6"
+  - "6"
```
Comment (kuisathaverat) [inline .ci/.jenkins_nodejs.yml]: Node.js 6 really? EOL Jul 2019

image

Comment (v1v) [reply .ci/.jenkins_nodejs.yml]: That's actually the list of versions that were declared in the .travis.yml, so no issues to remove it, but let's tackle that in a follow up if required :)

**Pull request merged**